### PR TITLE
Added unit tests for json parser

### PR DESCRIPTION
**Pull request by rsupnekar**

### CODE REVIEW

**Review by ee11cbb19052e40b07aac0ca060c23ee — APPROVED** [review]

File: sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    @Test
+    public void invalidJsonTest() {
+        Log.d(TAG, "Invalid json test");
+        try {
+            GeoJsonParser.parse(invalidJsonString);
+            Assert.fail("Invalid json gave no error");
+        } catch (Exception e) {
+            Log.d(TAG, "Invalid json threw exception, test passed");
+        }
```
Comment (empbarney) [inline sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java:102]: If you know what exception to expect then you can actually set up a test to get that exception. Some of the existing tests show how it is done OR it is there on the web.

File: sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        InputStream stream = null;
+        try {
+            stream = classLoader.getResourceAsStream("cmapi.json");
+        } catch (Exception e) {
+            Assert.fail("Could not get resource stream for parsing");
+        }
+        try {
+            List<IFeature> featureList = GeoJsonParser.parse(stream);
+            Assert.assertEquals(2, featureList.size());
+        } catch (Exception e) {
```
Comment (empbarney) [inline sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java]: Do you want to check for the feature type created?

File: sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    private static final IMilStdRenderer milStdRenderer = new MilStdRenderer();
+
+    static {
+        storageManager.setEventManager(eventManager);
+        eventManager.setStorageManager(storageManager);
+        coreManager.setStorageManager(storageManager);
+        coreManager.setEventManager(eventManager);
+        milStdRenderer.setStorageManager(storageManager);
+    }
+
```
Comment (empbarney) [inline sdk/sdk-view/src/test/java/mil/emp3/json/geoJson/GeoJsonParserTest.java:60]: Why do you need all the above stuff here?

**Comment by rsupnekar** [comment]
Yes.  Without StoreManager, it crashes while trying to instantiate a feature.  The parser is converting the string to a feature object.


Yes, I am adding more details to the parser test (your second comment).  Like I said, this was my starting point.

________________________________

**Comment by rsupnekar** [comment]
Fixes #28

**Pull request merged**